source/blog/2018-02-14-ember-3-1-beta-released.md

```diff
@@ -0,0 +1,159 @@
+`defineProperty`. In recent years native JavaScript setter and getter
+impementations have become fast and mature.
+
+Starting in Ember 3.1 (and described in [RFC
+281](https://github.com/emberjs/rfcs/blob/master/text/0281-es5-getters.md) you
+are now able to read the value of a computed property using a native ES5 getter.
+For example this component which uses computed properties:
+
+```js
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+
+  name: computed('firstName', 'lastName', function() {
+    return `${this.get('firstName')} ${this.get('lastName')}`;
+  }),
+
+  message: computed('name', function() {
+    return `Hello ${this.get('name')}!`;
+  });
+
+});
+```
+
+Can be re-written using ES5 getters:
+
+```js
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+
+  name: computed('firstName', 'lastName', function() {
+    return `${this.firstName} ${this.lastName}`;
+  }),
+
+  message: computed('name', function() {
+    return `Hello ${this.name}!`;
+  });
+
+});
+```
+
+Legacy `get` features are not deprecated or removed in 3.1. In fact there are
+several cases where you must still use `get`:
+
+* If you are calling `get` with a chained path. For example in `this.get('a.b.c')`
+if `b` is `undefined` the return value is `undefined`. Converting this
+to `this.a.b.c` when `b` is `undefined` would instead raise an exception.
+* If your object is using `unknownProperty` you must continue to use `get`.
+Using an ES5 getter on an object with `unknownProperty` will cause an
+assertion failure in development.
+* Ember Data returns promise proxy objects when you read an async relationship
+and from other API. Ember proxy objects, including promise proxies, still
+require that you call `get` to read values.
+
+With these caveats in mind, how should you know if you can convert a `get`
+call to a native getter? If you have code where `get` is called on `this` you
+likely can convert it. If you have a `get` on another object,
+`anything.get('foo')`, you should exercise caution when converting to a native
+getter.
+
+The community-provided
+[es5-getter-ember-codemod](https://github.com/rondale-sc/es5-getter-ember-codemod)
+is a great way to convert your existing codebase to ES5 getters. It follows
+the conservative guidelines and only converts `this.get`. Note that is cannot
+make all possible converstions to the new API, nor can it ensure 100% of the
+convertions it makes are correct. If your app has poor test coverage or you
+lack any confidence in your ability to make regression checks a manual
+and gradual conversion process may be more appropriate.
+
+#### Application Template Wrapper
+
+  - Application Template Wrapper (codemod built into @ember/optional-features)
+  - Enableable via: @ember/optional-features
+    - `ember feature:enable application-template-wrapper`
```

Comment:
I was trying this out today, and it seems that @ember/optional-features has application-template-wrapper **enabled** by default and thus requires _disabling instead of enabling_ to change the standard behavior.

Assuming that this is the intended behavior from @ember/optional-features, then this feature is **disableable** using `ember feature:disable application-template-wrapper`

Reply:
The behavior of the addon is correct.  The mental model here is that ember has long had a "feature" to wrap your entire applications contents in a `<div></div>`. If Ember were to remove that feature it would be a breaking change (because of styling or any number of other reasons). So instead of removing it, we have made the "wrap my application template contents in another element" feature _optional_.